src/manifolds/GraphManifold.jl

```diff
@@ -0,0 +1,89 @@
+
+using LightGraphs
+import LightGraphs: AbstractSimpleGraph
+
+@doc doc"""
+    GraphManifoldType
+
+This type represents the type of data on the graph that the [`GraphManifold`](@ref)
+represents.
+"""
+abstract type GraphManifoldType end
+
+@doc doc"""
+    EdgeGraphManifold <: GraphManifoldType
+
+A type for a [`GraphManifold`](@ref) where the data is given on the edges.
+"""
+struct EdgeGraph <: GraphManifoldType end
+
+@doc doc"""
+    VectexGraphManifold <: GraphManifoldType
+
+A type for a [`GraphManifold`](@ref) where the data is given on the vertices.
+"""
+struct VertexGraph <: GraphManifoldType end
+
+@doc doc"""
+    GraphManifold{G,M,T}
+
+Build a manifold, that is basically a [`PowerManifold`](@ref) either on the
+edges or vertices of a graph.
+
+# Fields
+* `G` is an `AbstractSimpleGraph`
+* `M` is a [`Manifold`](@ref)
+* `T` is a [`GraphManifoldType`](@ref), i.e. whether data is given on vertices
+or edges.
+"""
+struct GraphManifold{
+        G<:AbstractSimpleGraph,
+        M<:Manifold,
+        T<:GraphManifoldType
+    } <: Manifold
+    graph::G
+    manifold::M
+    type::T
+end
+
+@doc doc"""
+    check_manifold_point(M::GraphManifold,x)
+
+check whether `x` is a valid point on the [`GraphManifold`](@ref)
+"""
+check_manifold_point(::GraphManifold,::Any)
+
+function check_manifold_point(
+    M::GraphManifold{G,TM,VertexGraph},
+    x;
+    kwargs...) where {G <: AbstractSimpleGraph,TM<:Manifold}
+    for i in nv(M.graph)
```

Comment:
This could just call `check_manifold_point` on the `PowerManifold`. I guess a lot of operations could be just forwarded to `PowerManifold`.

Reply:
Of course, there could be some additional checks here, like whether the power manifold has the number of components that matches the graph.

Reply:
yeah, with the idea to the `PowerManifold` that sounds right, and the additional check is then just the matching size. Very neat. This also means a lot of functions are already implemented (in Power).